cpp/locality/NeighborQuery.h

```diff
@@ -37,16 +37,42 @@ struct NeighborPoint {
 
     //! Default comparator of points is by distance.
     /*! This form of comparison allows easy sorting of nearest neighbors by
-     *  distance
+     *  distance.
      */
     bool operator< (const NeighborPoint &n) const
         {
         return distance < n.distance;
         }
 
-    unsigned int id;
-    unsigned int ref_id;
-    float distance;
+    unsigned int id;            //! The point id.
+    unsigned int ref_id;        //! The reference point id.
+    float distance;             //! The distance between the point and the reference point.
+};
+
+
+//! (Almost) POD class to hold information about generic queries.
+/*! This class provides a standard method for specifying the type of query to
+ *  perform with a NeighborQuery object. Rather than calling queryBall
+ *  specifically, for example, the user can call a generic querying function and
+ *  provide an instance of this class to specify the nature of the query.
+ */
+struct QueryArgs {
+    //! Define constructor
+    /*! We must violate the strict POD nature of the class to support default
+     *  values for parameters.
+     */
+    QueryArgs() : nn(-1), rmax(-1), scale(-1), exclude_ii(false) {}
+
+    enum QueryType {
+        ball,         //! Query based on distance cutoff.
+        nearest       //! Query based on number of requested neighbors.
+    };
+
+    QueryType mode;     //! The number of nearest neighbors to find.
+    unsigned int nn;    //! The number of nearest neighbors to find.
```

Comment:
I'm a little confused at why this is an unsigned int here, I changed it locally already. I assume it will fix itself when I push all the other changes though.